### PR TITLE
Implement CRUD operations on user groups

### LINKED ISSUE #5: Implement CRUD operations on user groups
As an authenticated (logged-in) user, I want to create groups with other users and manage them by myself. They must support CRUD operations (create, read, update, delete). 

Every group has one owner and many users. Only the owner is able to update or delete the group.
As a developer, I want to have backend functionality for these operations. The user who is creating a group is automatically its owner. 

A lucidchart diagram for the database structure: https://lucid.app/documents/view/6dd7e3ee-b429-414b-a4d2-0969900a44f9

### Dependencies

Depends on:

- User management

### Acceptance Criteria
- [x] Read, Update and Delete return 401 for Groups one is not the Owner of
- [x] Create,Read, Update and Delete return 401 for GroupAssignments for groups one is not the owner of
- [x] Read + Update return 404 on both tables if the entity is not found
- [x] Read, Update and Delete return 401 for Groups one is not the Owner of
- [x] Create returns 201 on successful create and includes a link to the ressource
- [x] All undefined success cases return 200
- [x] Functions are developed for CRUD operations. There are Read operations for individual items as well as for collections
- [x] Ownership grants are implemented and taken into account

### PR DESCRIPTION
**Pull request by TobiPeterG**
This closes #5

### CODE REVIEW

**Comment by TobiPeterG** [comment]
Known: rubocop still doesn't run through completely.

File: spec/views/groups/new.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe "groups/new", type: :view do
+  before do
+    assign(:group, Group.new(
```
Comment (kbroschke) [inline spec/views/groups/new.html.erb_spec.rb:5]: Same as in spec/views/groups/edit.html.erb_spec.rb

**Review by kbroschke — CHANGES_REQUESTED** [review]
Notices generated by the controller are not internationalized and are shown twice. I put comments on the corresponding lines.
Additionally I was unsure with some parts of the tests. I also marked those spots with comments.

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<p style="color: green"><%= notice %></p>
```
[the remark this answers is not in the document]
Comment (kbroschke) [reply app/views/groups/index.html.erb]: Resolved by c6d919a

File: app/views/groups/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<p style="color: green"><%= notice %></p>
```
[the remark this answers is not in the document]
Comment (kbroschke) [reply app/views/groups/show.html.erb]: Resolved by c6d919a

File: app/controllers/groups_controller.rb [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+  # POST /groups or /groups.json
+  def create
+    @group = Group.new(group_params)
+
+    respond_to do |format|
+      if @group.save
+        membership = Membership.create(:user => current_user, :group => @group, :role => :admin)
+        membership.save()
+        format.html { redirect_to group_url(@group), notice: "Group was successfully created." }
```
[the remark this answers is not in the document]
Comment (kbroschke) [reply app/controllers/groups_controller.rb]: Resolved by e81d6e2

File: spec/requests/groups_spec.rb [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+      let(:new_attributes) do
+        { name: "Another Group" }
+      end
+
+      it "updates the requested group" do
+        group = Group.create! valid_attributes
+        membership = Membership.create!(:user => user, :group => group, :role => :admin)
+        patch group_url(group), params: { group: new_attributes }
+        group.reload
+        skip("Add assertions for updated state")
```
[the remark this answers is not in the document]
Comment (kbroschke) [reply spec/requests/groups_spec.rb]: Resolved by 6431b16

File: spec/requests/groups_spec.rb [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    it "renders a successful response" do
+      get new_group_url
+      expect(response).to be_successful
+    end
+  end
+
+  describe "GET /edit" do
+    it "renders a successful response" do
+      group = Group.create! valid_attributes
+      membership = Membership.create!(:user => user, :group => group, :role => :admin)
```
[the remark this answers is not in the document]
Comment (DieKautz) [reply spec/requests/groups_spec.rb]: should resolve with c45af72

File: spec/views/groups/edit.html.erb_spec.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe "groups/edit", type: :view do
+  let(:group) do
+    Group.create!(
```
[the remark this answers is not in the document]
Comment (DieKautz) [reply spec/views/groups/edit.html.erb_spec.rb]: should resolve with eb5b2c7

File: spec/views/groups/new.html.erb_spec.rb (hunk 1 above)
Comment (DieKautz) [reply spec/views/groups/new.html.erb_spec.rb]: should resolve with eb5b2c7

**Review by kbroschke — APPROVED** [review]

**Review by KiruChaff — APPROVED** [review]
Everything looks fine :)

**Pull request merged**